### ISSUE TITLE
Very different MCscan output when "dist" changes

### ISSUE THREAD

**Issue by evolgenomology** [tanghaibao/jcvi#229]
Hi,

I have a question regarding the dist parameter in my comparison of animal genomes. I am running the synteny search like so

`python -m jcvi.compara.catalog ortholog --dist=400 --cscore=0.7 --dbtype=prot --no_strip_names --dbtype=prot --full pecten emu`.

Usually I have been operating with dist=200, as the default seemed far to stringent for the evolutionary distances I am operating on. Now for one comparison I tried dist=200 and dist=400 and found that the plots look distinctly different. That is, there are blocks present in one, but missing in the other. I am guessing there is something I don't understand, or am not considering. It would be super great to get some feedback. Plots attached.

Cheers

Philipp
[pecten.emu.1x1.lifted.prot.pdf](https://github.com/tanghaibao/jcvi/files/4421690/pecten.emu.1x1.lifted.prot.pdf)
[pecten.emu.1x1.lifted.400.pdf](https://github.com/tanghaibao/jcvi/files/4421692/pecten.emu.1x1.lifted.400.pdf)

**Comment by evolgenomology** [comment]
As an addendum, and this is perplexing. I tested one of the species from above (E. muelleri = emu) against a different species with three distance settings, default, 200, 400. The results are here and in this case they are as I would expect them, i.e. more genes anchoring synteny for larger dist settings. Every idea what could be happening in the failing example above would be highly welcome.
[xbocki.emu.1x1.lifted.dist20.pdf](https://github.com/tanghaibao/jcvi/files/4422232/xbocki.emu.1x1.lifted.dist20.pdf)
[xbocki.emu.1x1.lifted.dist400.pdf](https://github.com/tanghaibao/jcvi/files/4422238/xbocki.emu.1x1.lifted.dist400.pdf)

[xbocki.emu.1x1.lifted.pdf](https://github.com/tanghaibao/jcvi/files/4422234/xbocki.emu.1x1.lifted.pdf)

**Comment by tanghaibao** [comment]
@evolgenomology 

You are using a `dist` value that is huge! The conservation signal that you are seeing, if any, is super week possibly due to the evolutionary distance.

Normally the `--dist=200` should generate a strict subset of the `--dist=400` results, since basically the latter allows you to recruit more anchor points into the blocks and nothing more. However, your output file names suggested that you had turned on the option `--quota=1x1` which is analogous to getting the reciprocal best block, that's causing some blocks to be flipped back and forth due to equally weak scores when compared against each other. If you do not use the quota option, I believe that `--dist=200` generates a subset of `--dist=400`.

For your use case, I think not setting the quota would be the least confusing option, then you just experiment with the `dist`. From smaller to bigger, it should be clear that there an increasing level of signal, but signal from smaller dist should always be there.

Haibao

**Comment by evolgenomology** [comment]
@tanghaibao 

Thanks for the explanation. No, I am not setting this option `--quota=1x1`. Ah, now I see that this is the `--full` option. Hmmm, I actually assumed that it would be more specific to go for the 1x1, or reciprocal best block, in particular over the large evolutionary distances.
In these comparisons, between animal phyla, the smaller distance settings are far too stringent. I have tested this to some degree by trying to replicate what has been described here: https://www.nature.com/articles/nature16150. Basically to see what is shown in Fig. 3 I had to go up to 200. Now I was experimenting with larger distance settings, as I saw that in this comparison between pecten and sponge there wasn't much to be found. As you can see from the second set of plots, X. bocki (basal bilaterian) to the sponge, a dist=200 works quite well (while dist=20) is too stringent and also dist=200 is a subset of 400. So that's nice.

To finish with a question: you are basically saying running it without the `--full` option (no RBH!) is better? Is it better in every case for the large distances?
I am asking because this goes back to my previous Issue with the triplet comparison. There I noted that not in all cases blocks in A and B, and A and C, are also blocks in B and C, which is a bit worrying.

This would be doing this for the first example:
[pecten.emu.lifted.pdf](https://github.com/tanghaibao/jcvi/files/4422379/pecten.emu.lifted.pdf)

Cheers

Philipp

**Comment by tanghaibao** [comment]
@evolgenomology 

It is the combination of **distance** and **minimum size** cutoff that determines what gets to be shown. To understand how the synteny blocks are generated, you should understand that `dist` is the clustering distance, so everything within that distance will get recruited into the same block. Then the block needs to have a minimum number `n` of pairs, the default is `4`.

Now when you set a small distance, say `--dist=20`, a lot of the signal is not showing up because most of the clusters are 'singleton' clusters with no nearby points. When you increase it to `--dist=400`, more clusters reach the threshold of 4 pairs so you are seeing more.

To have a similar visual effect of Figure 3 in the paper, you can try to run with:

```console 
python -m jcvi.compara.catalog ortholog -n 1
```

This basically also outputs 'singleton clusters', which is in effect equivalent to visualizing the BLAST results. Of course, in this case, you are not really clustering anything, so you can get around with not setting the `--dist` (so all distance settings work).

As soon as you turn on clustering, you need to understand how `--dist` and `-n` work together in order to predict what's going to happen.

Also, I think you don't need to use quota filtering (don't use `--full`) in your case since filtering `1x1` on the block level will remove some of the signal you are looking for.

PS: I was not exposing the minimum size (`-n`) in the `ortholog` command before, which I fixed in 56bc37b11a6 just now. Update your `jcvi` package first if you want to play with this option.

**Comment by evolgenomology** [comment]
@tanghaibao

Thanks again for the explanation. Yes, I was actually quite happy that it is so easy to play with the distance parameter here. Previously, I had used other methods and found that for my species that I had to be rather lenient with the minimum number of pairs. I think with the evolutionary distances I am dealing with, there are simply too many re-arrangements. The comparison of Saccoglossus and Amphioxus, as shown in the Fig 3 of the paper, is >500 Ma of independent evolution. Actually, I was a bit critical of the detour they kind of took in that paper to come to their synteny plot. Mainly they had to do this of course, because the Saccoglossus genome is still quite fragmented.
Not saying that what they show is wrong, I could confirm it by running MCscan in your pipeline and the micro-synteny they describe later on is also there.

I will try the `-n 1` option, but I am not sure that is what I want. Or well, it will be interesting to see the singleton clusters, but what I really want to see is the larger conserved blocks between chromosomes. And I want to do this in a three or four way comparison under the assumption that even between very distant animal genomes it is the same genes that are “clustered” (= on roughly the same chromosomes).

Cheers

Philipp
